package/network/config/ltq-vdsl-app/files/dsl_control

```diff
@@ -194,34 +195,39 @@ start_service() {
 
 	eval "xtse=\"\${xtse_xdsl_$annex}\""
 
-	[ -z "${xfer_mode}" ] && xfer_mode=ptm
-
 	case "${xfer_mode}" in
 	atm)
-		LOAD=ltq_atm_vr9
-		UNLOAD=ltq_ptm_vr9
+		tc_layer="1:0x1:0x1_1:0x1:0x1"
 
 		# in most cases atm is used on top of adsl
 		[ -z "${line_mode}" ] && line_mode=adsl
 		;;
-	*)
-		LOAD=ltq_ptm_vr9
-		UNLOAD=ltq_atm_vr9
+	ptm)
+		tc_layer="2:0x1:0x1_2:0x1:0x1"
 
 		# in most cases ptm is used on top of vdsl
 		[ -z "${line_mode}" ] && line_mode=vdsl
 		;;
+	*)
+		tc_layer="1:0x1:0x1_2:0x1:0x1"
+		;;
 	esac
 
 	case "${line_mode}" in
 	adsl)
 		mode=1
 
-		# mask out VDSL bits when ATM is requested
+		# mask out VDSL bits when ADSL is requested
 		xtse="${xtse%_*}_00"
 		;;
-	*)
+	vdsl)
 		mode=2
+
+		# mask out ADSL bits when VDSL is requested
+		xtse="00_00_00_00_00_00_00_${xtse##*_}"
```

Comment:
Do we really need to mask out the ADSL bits here? Do you encounter any incompatibilities with DSLAMs? Does the ADSL fall back for vectoring enabled lines + a **not** vectoring xdsl firmware still works with the ADSL bits masked? Nevertheless, this change should go into an extra commit. Since it is unrelated to the commit topic.